### ISSUE TITLE
[P1][PAPER/EXECUTION][PREPARED_WAIT] Canonical single-leg LAY liability and settlement authority

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/Autosport#1626]
Refs #1 whole product, #355 live/outcome-independent program, #619/#623 PAPER execution reality, #642/#646 runtime adoption, #1294 exchange-side quote identity, #1581 side-aware PaperValueAgent.

WORK_PACKET_ID=WP-E03-child-paper-single-leg-lay-liability-settlement
RUN_ID=GPT56SOL-20260922T1715CEST-PAPER-LAY-LIABILITY
SEMANTIC_KEY=paper.single-leg-lay-liability-settlement-v1
AUTHORITY_FAMILY=paper.execution-liability-settlement
PRIORITY=P1
MODE=PREPARED_WAIT_NO_SOURCE_MUTATION
DISPATCH_STATE=PREPARED_WAIT_FOR_CANONICAL_EXCHANGE_SIDE_BASE

## Product gap

The canonical #623 PAPER execution-reality model intentionally fails closed for non-BACK until a canonical LAY liability authority exists. Canonical #646 adoption likewise hardcodes ExecutionAction(side="BACK") because PaperBook currently treats ticket.stake as both order stake and capital debited/reserved, and its settlement law is BACK-only.

#1581 now repairs the upstream LAY EV sign and deliberately withholds positive-EV LAY after risk rather than laundering it into the BACK-only execution bridge. That is the correct current safety state, but it means PAPER cannot yet execute a legitimate positive-EV single-leg LAY.

This packet closes exactly that missing economic/runtime seam. It must not be implemented as a simple side-field propagation.

## Canonical economic law

For one single-leg LAY at accepted decimal odds O and accepted lay stake S:
- lay stake S is the amount won gross when the laid selection loses;
- liability / capital at risk L = S * (O - 1);
- if the laid selection loses: gross profit = +S;
- if the laid selection wins: gross loss = -L;
- if void: locked liability is returned with zero gross P&L.

Paper balance/reserve mechanics must therefore keep **order/execution stake** distinct from **capital debited/reserved**.

Example, bankroll 100, LAY stake 10 @ 5.0:
- liability = 40;
- open balance = 60 after reserving liability;
- selection loses -> settlement credit = liability + lay stake = 50 -> balance 110;
- selection wins -> settlement credit = 0 -> balance 60;
- void -> settlement credit = liability = 40 -> balance 100.

Commission/tax is NOT invented in this packet. Preserve existing canonical cost/commission authorities; base PaperBook settlement is gross before those external costs unless a canonical integrated cost seam already requires otherwise.

## Exact bounded scope after unblock

1. **Side-aware exposure primitive**
   - one exact Decimal helper/contract for BACK locked capital = stake and LAY locked capital = stake*(odds-1);
   - reject non-finite/non-positive odds/stake and unsupported side exactly;
   - no ambient Decimal-context dependence.

2. **#623 PAPER execution reality**
   - preserve ExecutionAction.requested_stake / PaperLegAttempt.execution_stake as order/matched stake semantics;
   - compute ACCEPTED/PARTIAL known exposure from exact accepted stake + accepted odds + side;
   - for LAY UNKNOWN, conservative exposure may be issued only when a canonical upper odds/liability bound exists. Otherwise fail closed rather than treating requested stake as liability;
   - synthetic/configured LAY execution must not invent favorable odds/slippage semantics. If no exact LAY execution model/evidence exists, allow canonical empirical/configured evidence only or stay fail-closed;
   - BACK behavior byte/semantic compatible.

3. **#646 adoption/materialization**
   - consume stable exchange_side from canonical quote/intent identity;
   - preserve side into ExecutionAction, durable attempt identity, PaperTicket/TicketLeg materialization and restart matching;
   - never infer LAY from labels after the fact;
   - only ACCEPTED/PARTIAL exact attempt truth may create PAPER exposure.

4. **PaperBook**
   - ticket persistence must distinguish execution/order stake from locked capital/liability; do not overload one field ambiguously;
   - opening a LAY debits/reserves liability, not lay stake;
   - settlement follows the gross laws above;
   - snapshot/lifecycle replay recomputes the same side-aware debit + settlement exactly;
   - old snapshots may be interpreted as BACK only if their schema/version contract proves that historical invariant; never silently reinterpret legacy records as LAY.

5. **Initial supported surface**
   - single-leg LAY only;
   - mixed BACK/LAY multi-leg/parlay remains fail-closed;
   - no claim of exchange cross-matching, commission exactness, real fill probability, provider write, or real-money readiness.

## Deterministic falsifiers

Minimum regression matrix:
- BACK existing controls unchanged;
- LAY S=10 @ O=1.5 => liability 5;
- LAY S=10 @ O=2 => liability 10;
- LAY S=10 @ O=5 => liability 40;
- accepted LAY open/reserve then win-selection, lose-selection, and VOID balance outcomes match exact laws;
- PARTIAL LAY matched stake 4 @ 5 => liability 16, not 4 and not requested full liability;
- LAY requested/accepted odds change recomputes liability from accepted odds for accepted exposure;
- UNKNOWN LAY without exact maximum-liability/odds bound cannot mint finite safe exposure;
- restart/snapshot round-trip preserves side, execution stake, liability and no duplicate debit/materialization;
- duplicate/redelivered accepted attempt cannot double-reserve;
- mixed-side or multi-leg LAY remains rejected in this first packet;
- commission/tax is not silently subtracted or assumed zero-authoritative.

## Dependencies / unblock

WAIT until:
- #1294/#1581 (or a canonical integrated successor) establishes stable exchange_side through quote -> PaperValue decision/risk identity; and
- exact current-main #623/#646 descendant topology is reread before mutation.

When unblocked, continue the existing #623/#646/PaperBook authorities; do not create a second execution ledger or second PaperBook.

Likely touched authorities after unblock:
- src/autosport/paper_execution_reality.py
- src/autosport/paper_execution_adoption.py
- src/autosport/paper.py
- canonical TicketLeg/PaperTicket domain schema only as required
- focused tests for exact liability, settlement, partial fill and restart

EXCLUDES=provider writes; real-account execution; Betfair live liability authority; provider commission/tax; arbitrary multi-leg LAY; portfolio/risk-policy replacement; readiness/release/HUMAN/NVDA promotion.

DONE_WHEN=A positive canonical single-leg LAY can traverse side-aware risk -> #623 attempt -> #646 materialization -> PaperBook reserve/settlement/restart with exact liability semantics, while unsupported/unknown LAY paths remain fail-closed and BACK regressions stay clean.

REAL_MONEY_EXECUTION=false
HUMAN_TESTED=false
NVDA_VERIFIED=false
WHOLE_PRODUCT_COMPLETE=false

**Comment by Oleksii-debug** [comment]
DEPENDENCY_REFRESH_V1
UPSTREAM_PR=#1581
UPSTREAM_CURRENT_HEAD=e71e00693c183cf8d805ee4f780eaa46cad98a2a
PREVIOUS_ROUTED_HEAD=fbc6242a1dffa803babfebdacc04a0d423cba836
MATERIAL_CHANGE=#1581 now also binds the risk-policy amount for LAY to maximum-loss liability: chosen order stake remains the lay-stake unit, while PaperRiskPolicy receives chosen_stake*(odds-1). The pre-#623 LAY execution fence remains intact.

#1626 SCOPE_CORRECTION:
- do NOT duplicate upstream EV or pre-execution risk-liability sizing;
- begin at the next boundary: #623 execution-reality exposure, #646 side-preserving materialization, and PaperBook locked-capital/settlement/restart semantics;
- accepted/partial exposure liability must use accepted execution stake + accepted execution odds;
- unsupported UNKNOWN upper-liability and mixed/multi-leg LAY remain fail-closed.

UNBLOCK remains an integrated/stable canonical exchange-side base plus fresh #623/#646 topology reread.
REAL_MONEY_EXECUTION=false
HUMAN_TESTED=false
NVDA_VERIFIED=false
WHOLE_PRODUCT_COMPLETE=false

**Comment by Oleksii-debug** [comment]
CLAIM_V2
RUN_ID=GPT56SOL-20260923T0134+0200-1626-EXEC-LIABILITY
SEMANTIC_KEY=paper.single-leg-lay-liability-execution-v1
AUTHORITY_FAMILY=paper.execution-liability-settlement
INTENDED_SLICE=src/autosport/exchange_exposure.py + src/autosport/paper_execution_reality.py + focused tests only. Exact behavior: canonical BACK/LAY exact Decimal locked-capital primitive; #623 execution-reality accepted/partial exposure uses accepted execution stake+odds+side; synthetic/configured LAY and unsupported UNKNOWN LAY fail closed; BACK behavior remains unchanged. No PaperBook/adoption/domain source mutation in this child.
MODE=SOURCE_MUTATION
FINAL_COLLISION_CHECK=#1626 comments contain no newer mutating claim; no open PR matches parent semantic key; main advanced only by unrelated BETDAQ account files to 86203fa3b0a55dc039d8ee14e87791b226908083, so the selected source slice remains non-overlapping.
REAL_MONEY_EXECUTION=false
HUMAN_TESTED=false
NVDA_VERIFIED=false
WHOLE_PRODUCT_COMPLETE=false

**Comment by Oleksii-debug** [comment]
TASK_RESULT_V2
RUN_ID=GPT56SOL-20260923T0134+0200-1626-EXEC-LIABILITY
SEMANTIC_KEY=paper.single-leg-lay-liability-execution-v1
AUTHORITY_FAMILY=paper.execution-liability-settlement
RESULT=SOURCE_IMPLEMENTED_PR_OPEN_BLOCKED_ON_RECONVERGENCE
PR=#1725
HEAD_SHA=41f080ed0c8b6c150dcc69e38eb5654ea0b46baf
FILES=src/autosport/exchange_exposure.py; src/autosport/paper_execution_reality.py; tests/test_lay_execution_liability.py
IMPLEMENTED=exact Decimal BACK/LAY locked-capital primitive; ACCEPTED/PARTIAL empirical single-leg LAY execution exposure from accepted odds+stake+side; durable replay/resume derives identical liability; synthetic/configured/UNKNOWN/mixed LAY fail closed before run reservation; BACK stake exposure preserved.
LOCAL_PROPERTY_EVIDENCE=1000 randomized exact-Decimal LAY identities verified: liability == stake*(odds-1) == stake*odds-stake, plus deterministic BACK/LAY examples. This is math-only evidence, NOT repository test-suite evidence.
ROUTING_BLOCKER=GitHub read/compare endpoints returned 403 rate-limit after publication. PR creation succeeded and reports base main a0ca6003d2326ad099f70979d40c0333c19388be, head 41f080ed0c8b6c150dcc69e38eb5654ea0b46baf, changed_files=3, mergeable=false. Branch originated at older main 86203fa3b0a55dc039d8ee14e87791b226908083. Do not merge/force-update until actual main overlap is readable and reconverged. CI PASS is NOT claimed.
CLAIM_REMAINS_ACTIVE=true
NEXT=when GitHub reads recover, inspect main changes to paper_execution_reality.py, reconverge #1725 source semantics onto current main, run focused/full CI, then disposition. No PaperBook/adoption mutation is owned by this claim.
REAL_MONEY_EXECUTION=false
HUMAN_TESTED=false
NVDA_VERIFIED=false
WHOLE_PRODUCT_COMPLETE=false

**Comment by Oleksii-debug** [comment]
CHILD_HANDOFF_V2
RUN_ID=GPT56SOL-20260923T0132Z-PAPERBOOK-LAY-FENCE
PARENT=#1626
CHILD_PR=#1789
SEMANTIC_KEY=paper.exchange-side-lay-materialization-fail-closed-v1
AUTHORITY_FAMILY=paper-execution.economic-materialization
EXACT_HEAD_SHA=6a2107a6ee8227a2554366591e71266f06c04213
STACK_BASE=#1630@2b669d3bbde3daedaf64549ecaa23006ade2c5f7
RESULT=INTERIM_ROOT_FENCE_SOURCE_FROZEN
MATERIAL_EFFECT=Until #1626 integrates full side-aware PaperBook liability/settlement, direct open/save/schema-8 load cannot turn LAY identity into trusted BACK-only bankroll/settlement state. LAY remains valid at MarketEvent/TicketLeg identity layer. BACK and legacy None behavior remain.
FILES=src/autosport/paper.py;tests/test_exchange_side_ticket_identity.py
GATES=CI/Windows Candidate/Endurance queued on exact head; queued is not PASS.
NO_OVERLAP=#1725 active execution-liability child remains untouched; it explicitly excludes PaperBook/adoption/domain.
LOCAL_TESTS=NOT_RUN__worker checkout DNS failure.
REAL_MONEY_EXECUTION=false
HUMAN_TESTED=false
NVDA_VERIFIED=false
WHOLE_PRODUCT_COMPLETE=false
